### PR TITLE
Taylor - Signup Form Component

### PR DESCRIPTION
**Pull request by TaylorDJones11**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other (please specify here: Signup Form Component)
  
  
## Description of what this Pull Request does
- The Signup Form component should accept the users name/email/password after hitting the "SIGN UP" button
- The Signup Form component will clear after each submission
- The Signup Form is validated
- The Signup Form component will push users to Firebase database

## Related Tickets & Documents
- Closes  https://github.com/AritDeveloperCircle/findyourpeople/issues/58
- Related Issues: https://github.com/AritDeveloperCircle/findyourpeople/issues/31
  
## QA Instructions, Screenshots, Recordings
Screenshot 2023-04-28 at 21 16 56

  
## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [x] I need help with tests
- [ ] No, and here is why: __explain here__

### CODE REVIEW

**Review by annonymousauthority — APPROVED** [review]
Not sure if the images used are meant for this type of app...

Maybe it can be reviewed for a more fitting use case.

Overall ok.

**Comment by TaylorDJones11** [comment]
> Not sure if the images used are meant for this type of app...
> 
> Maybe it can be reviewed for a more fitting use case.
> 
> Overall ok.

I was just using it for a placeholder image.

File: src/components/common/signupform/signupform.module.css [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  width: 90vw;
+  max-width: 600px;
+  background: #fff;
+  border-radius: 0.25rem;
+  box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
+    0 2px 4px -1px rgba(0, 0, 0, 0.06);
+  padding: 2rem 2.5rem;
+  margin: 3rem auto;
+}
+.form_label {
```
Comment (msarit) [inline src/components/common/signupform/signupform.module.css:30]: CSS properties should be `kebab-case` not `snake_case` 🙏🏾

File: src/components/common/signupform/signupformcontainer.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          width={100}
+          src={images.signupAvatar}
+          alt='main avatar'
+        />
+      </div>
+      <h2 className={styles.header}>Create an account</h2>
+      <h4 className={styles.sub}>
+        Lets get started with creating your community!
+      </h4>
+      <SignupForm />
```
Comment (msarit) [inline src/components/common/signupform/signupformcontainer.js:22]: I believe our wireframes indicated an `<Aside>` component to be side-by-side with the SignUp form.

That component will be available with @Sheyitofunmi's [PR](https://github.com/AritDeveloperCircle/findyourpeople/pull/46)

Please leave a comment here to indicate that intention. 👍🏾

File: src/components/common/signupform/signupformcontainer.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import Link from 'next/link'
+import styles from './signupcontainer.module.css'
+import CustomizableAvatar from '../CustomizableAvatar'
+import SignupForm from './SignupForm'
+import images from '../../../images/index'
```
Comment (msarit) [inline src/components/common/signupform/signupformcontainer.js:5]: I believe the `images` you are importing here is capitalized "I" 🤔

**Review by msarit — CHANGES_REQUESTED** [review]
Great work @TaylorDJones11 

These are my preliminary review comments.

I have not yet QA'd on my `local` app, because we do not hav `CustomizableButton` yet.

File: src/components/common/signupform/signupformcontainer.js (hunk 2 above)
Comment (TaylorDJones11) [reply src/components/common/signupform/signupformcontainer.js:22]: > I believe our wireframes indicated an `<Aside>` component to be side-by-side with the SignUp form.
> 
> That component will be available with @Sheyitofunmi's [PR](https://github.com/AritDeveloperCircle/findyourpeople/pull/46)
> 
> Please leave a comment here to indicate that intention. 👍🏾

Yes, I'm just waiting for her PR to come in to complete the full signup page

File: src/components/common/signupform/SignupForm.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+          <input
+            type='password'
+            className={styles.form_input}
+            id='password'
+            name='password'
+            value={user.password}
+            onChange={handleChange}
+            required
+            minLength={6}
+            maxLength={20}
```
[the remark this answers is not in the document]
Comment (TaylorDJones11) [reply src/components/common/signupform/SignupForm.js:94]: > These password validations are good; I think we need some validations around password complexity?

Not sure what you mean here. Are you saying to add in regex here as I did for the email pattern?
Screenshot 2023-05-03 at 13 53 08

**Comment by TaylorDJones11** [comment]
@msarit I'm still having issues with Github. Can I delete this PR/create a new branch for this component? I've tried many things and recreating this PR seems like the best option.

File: src/components/common/signupform/SignupForm.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (msarit) [reply src/components/common/signupform/SignupForm.js:94]: Yes; I am thinking of checking the absence of repeated characters (for example) or the presence of special symbols.
[
I found ](https://www.oreilly.com/library/view/regular-expressions-cookbook/9781449327453/ch04s19.html )this cool resource that shows how o generate the regexes and the function that applies the regexes.

**Comment by msarit** [comment]
> @msarit I'm still having issues with Github. Can I delete this PR/create a new branch for this component? I've tried many things and recreating this PR seems like the best option.

@TaylorDJones11 Alright Taylor, go for it!

**Pull request closed without merging**